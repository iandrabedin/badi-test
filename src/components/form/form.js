import React from "react";
import "./form.scss";

export const Form = props => {
  const { searchIngredient, handleChangeIngredient, onSubmit } = props;

  return (
    <form className="form" onSubmit={onSubmit}>
      <div className="form-input">
        <label htmlFor="ingredient">Ingredients</label>
        <input
          name="ingredient"
          id="ingredient"
          type="text"
          placeholder="Onions, garlic..."
          value={searchIngredient}
          onChange={handleChangeIngredient}
        />
      </div>
      <button className="button" name="button-submit" type="submit">
        Search
      </button>
    </form>
  );
};
