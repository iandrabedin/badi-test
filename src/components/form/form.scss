@import "../../styles/variables.scss";

.form {
  display: flex;
  flex-wrap: wrap;
  margin: 16px 0;
  align-items: center;
  padding-bottom: 14px;

  &-input {
    width: 100%;
    min-height: 80px;
  }

  label {
    display: block;
    font-size: $font-size-small;
    color: $color-grey-dark;
  }

  input {
    padding: 12px 16px 14px;
    border-radius: 0.25rem;
    border: 1px solid rgba(33, 37, 41, 0.15);
    width: -webkit-fill-available;
  }
}

// Tablet - Desktop
@media (min-width: 768px) {
  .form-input {
    width: 30%;
  }

  button {
    margin: 0 32px;
  }
}
