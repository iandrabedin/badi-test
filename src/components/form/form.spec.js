import React from "react";
import { render, cleanup, fireEvent } from "@testing-library/react";
import { Form } from "./form";

afterEach(cleanup);

const props = {
  searchIngredient: "milk",
  handleChangeIngredient: jest.fn(),
  onSubmit: jest.fn()
};

describe("Form", () => {
  it("should render as expected", () => {
    const { asFragment } = render(<Form {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should have a submit button", () => {
    const { getByText } = render(<Form {...props} />);
    expect(getByText("Search")).toHaveAttribute("type", "submit");
  });

  it("should call submit function on click", () => {
    const onSubmit = jest.fn(e => e.preventDefault());
    const { getByText } = render(<Form {...props} onSubmit={onSubmit} />);
    fireEvent.click(getByText("Search"));
    expect(onSubmit).toHaveBeenCalledTimes(1);
  });
});
