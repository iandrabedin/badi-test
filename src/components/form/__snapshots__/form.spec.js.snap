// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should render as expected 1`] = `
<DocumentFragment>
  <form
    class="form"
  >
    <div
      class="form-input"
    >
      <label
        for="ingredient"
      >
        Ingredients
      </label>
      <input
        id="ingredient"
        name="ingredient"
        placeholder="Onions, garlic..."
        type="text"
        value="milk"
      />
    </div>
    <button
      class="button"
      name="button-submit"
      type="submit"
    >
      Search
    </button>
  </form>
</DocumentFragment>
`;
